scripts/approximations/main.py

```diff
@@ -0,0 +1,90 @@
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+
+import polynomial
+import rational
+import approximations
+
+# This script does the following:
+# - Computes polynomial and rational approximations of a given function (e^x by default).
+# - Computes (x,y) coordinates for every approximation given the same x coordinates.
+# - Plots the results for rough comparison.
+# The following are the resources used to write the script:
+# https://xn--2-umb.com/22/approximation/
+# https://sites.tufts.edu/atasissa/files/2019/09/remez.pdf
+def main():
+
+    ##############################
+    # 1. Configuration Parameters
+
+    # start of the interval to calculate the approximation on
+    x_start = 0
+    # end of the interval to calculate the approximation on
+    x_end = 3
+    # number of terms in the polynomial approximation / numerator of the rational approximation.
+    num_terms_approximation = 3
+
+    # number of (x,y) coordinates used to plot the resulting approximation.
+    num_points_plot = 10
```

Comment:
Hrmm, this should be way higher / so dense we don't notice, no?

Reply:
I chose a lower number to see the difference in accuracy between each of the approximations. But it could be anything, depending on what is your goal for visualizing it

Reply:
I think this should be way higher, we should be able to see the accuracy diff with more points as well, no?

Perhaps lets set to like 10000?

Reply:
Went even higher with `100000000`:

```
Max Error on [0, 1]
100000000 coordinates equally spaced on the X axis
3 polynomial terms and (3, 3) rational terms used for approximations.


Equispaced Poly: 0.014420666050105169
Chebyshev Poly: 0.009865548570433536
Chebyshev Rational: 7.428807818232741e-06